src/blockrelay/graphene.cpp

```diff
@@ -431,42 +431,31 @@ bool CGrapheneBlock::HandleMessage(CDataStream &vRecv, CNode *pfrom, std::string
         }
     }
 
-    bool result = grapheneBlock.process(pfrom, nSizeGrapheneBlock, strCommand);
+    bool result = grapheneBlock->process(pfrom, strCommand, pblock);
 
     return result;
 }
 
-bool CGrapheneBlock::process(CNode *pfrom,
-    int nSizeGrapheneBlock,
-    std::string strCommand) // TODO: request from the "best" txn source not necessarily from the block source
+bool CGrapheneBlock::process(CNode *pfrom, std::string strCommand, std::shared_ptr<CBlockThinRelay> pblock)
 {
     // In PV we must prevent two graphene blocks from simulaneously processing from that were recieved from the
     // same peer. This would only happen as in the example of an expedited block coming in
     // after an graphene request, because we would never explicitly request two graphene blocks from the same peer.
     if (PV->IsAlreadyValidating(pfrom->id))
         return false;
 
-    graphenedata.ClearGrapheneBlockData(pfrom);
-    pfrom->nSizeGrapheneBlock = nSizeGrapheneBlock;
+    std::shared_ptr<CGrapheneBlock> grapheneBlock = pblock->grapheneblock;
 
```

Comment:
validate grapheneBlock != nullptr

Reply:
also is this == pblock->grapheneblock?

I think this routine sometimes uses this and sometimes grapheneBlocks and may assume they are the same.  One idea is to DbgAssert(this == pfrom->grapheneBlock, return false)

Reply:
I'll take out graphenBlock, it's a bit of a circular reference...we only need to use "this"...that will also get rid of need to check for nullptr in this particular case.

Reply:
added checks for != nullptr and took out the references to "this" and just standardized across all thinrelay types in the same fashion.